### PR TITLE
add bs-seq-primer doc

### PR DESCRIPTION
**Pull request by sateeshperi**
Add primer doc on BS-Seq

### CODE REVIEW

File: docs/bs-seq-primer.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+1. **Reduced DNA Complexity**:
+
+   - Due to the conversion of cytosines to thymines, the DNA code becomes less diverse, increasing the likelihood of ambiguous alignments
+
+2. **Multiple DNA Strands**:
+
+   - The presence of four possible DNA strands (and their combinations) for each genomic locus increases the alignment search space
+
+3. **Variable Methylation States**:
+   - Each read can theoretically represent any possible methylation state for a locus, further complicating the alignment process
```
Comment (jfy133) [inline docs/bs-seq-primer.md:39]:
```suggestion
1. **Reduced DNA Complexity**:
   - Due to the conversion of cytosines to thymines, the DNA code becomes less diverse, increasing the likelihood of ambiguous alignments
2. **Multiple DNA Strands**:
   - The presence of four possible DNA strands (and their combinations) for each genomic locus increases the alignment search space
3. **Variable Methylation States**:
   - Each read can theoretically represent any possible methylation state for a locus, further complicating the alignment process
```

File: docs/bs-seq-primer.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+## **Applications of Bisulfite Sequencing**
+
+Bisulfite sequencing is employed to study:
+
+- **Genome-wide DNA methylation patterns**: Understanding epigenetic regulation of genes
+- **Differential methylation**: Investigating differences between different samples (e.g. healthy versus diseased states)
+- **Epigenetic inheritance**: Studying methylation changes across generations
```
Comment (jfy133) [inline docs/bs-seq-primer.md:47]: I would personally put this between principle and  and challenges

File: docs/bs-seq-primer.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+---
```
Comment (jfy133) [inline docs/bs-seq-primer.md:49]: Purpose of this?

File: docs/bs-seq-primer.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+## Three-Base Aligners: Bismark and BWA-Meth
+
+Bisulfite sequencing converts unmethylated cytosines into uracils (later read as thymines during PCR), leading to a DNA code represented predominantly by three bases (A, G, and T).
+
+Aligning these three-base reads against a standard reference genome is non-trivial, as the original strand identity and methylation state are obscured by the conversion.
+
+To address these challenges, specialized “three-base aligners” have been developed to accurately map bisulfite-treated reads and infer their original strand context and methylation status. Here, we primarily highlight the aligners used in the nf-core/methylseq pipeline.
```
Comment (jfy133) [inline docs/bs-seq-primer.md:57]:
```suggestion
To address these challenges, specialized “three-base aligners” have been developed to accurately map bisulfite-treated reads and infer their original strand context and methylation status. Here, we primarily summarise the aligners used in the nf-core/methylseq pipeline.
```

File: docs/bs-seq-primer.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+### Bismark ([docs](https://felixkrueger.github.io/Bismark/); [publication](https://pmc.ncbi.nlm.nih.gov/articles/PMC3102221/))
+
+- Bismark resolves strand ambiguity by performing up to four parallel alignments
+
```
Comment (jfy133) [inline docs/bs-seq-primer.md:62]:
```suggestion
```

File: docs/bs-seq-primer.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+- First, sequencing reads are converted _in-silico_ to represent both forward and reverse strand conversions (C-to-T and G-to-A), mirroring fully bisulfite-converted versions of the reference genome
+
```
Comment (jfy133) [inline docs/bs-seq-primer.md:64]:
```suggestion
```

File: docs/bs-seq-primer.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+- Each read set is then aligned using Bowtie2 (alternatively HISAT2 or minimap2) against equally converted references; this enables support for indels, local alignments, and bisulfite converted RNA-seq-type or long reads (e.g. EM-seq using Nanopore or Pac Bio reads)
+
```
Comment (jfy133) [inline docs/bs-seq-primer.md:66]:
```suggestion
```

File: docs/bs-seq-primer.md [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+Aligning these three-base reads against a standard reference genome is non-trivial, as the original strand identity and methylation state are obscured by the conversion.
+
+To address these challenges, specialized “three-base aligners” have been developed to accurately map bisulfite-treated reads and infer their original strand context and methylation status. Here, we primarily highlight the aligners used in the nf-core/methylseq pipeline.
+
+### Bismark ([docs](https://felixkrueger.github.io/Bismark/); [publication](https://pmc.ncbi.nlm.nih.gov/articles/PMC3102221/))
+
+- Bismark resolves strand ambiguity by performing up to four parallel alignments
+
+- First, sequencing reads are converted _in-silico_ to represent both forward and reverse strand conversions (C-to-T and G-to-A), mirroring fully bisulfite-converted versions of the reference genome
```
Comment (jfy133) [inline docs/bs-seq-primer.md:63]:
```suggestion
- First, sequencing reads are converted _in silico_ to represent both forward and reverse strand conversions (C-to-T and G-to-A), mirroring fully bisulfite-converted versions of the reference genome
```

File: docs/bs-seq-primer.md [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+- Each read set is then aligned using Bowtie2 (alternatively HISAT2 or minimap2) against equally converted references; this enables support for indels, local alignments, and bisulfite converted RNA-seq-type or long reads (e.g. EM-seq using Nanopore or Pac Bio reads)
+
+- By comparing these (up-to) four alignments, Bismark identifies each read’s correct strand origin
+
+This approach allows Bismark to handle directional, PBAT, amplicon, and non-directional libraries robustly and to accurately align reads that represent partially methylated cytosines.
+
+### BWA-Meth ([docs](https://github.com/brentp/bwa-meth); [publication](https://arxiv.org/abs/1401.1129))
+
+- BWA-Meth adapts the BWA-MEM algorithm for bisulfite data, providing efficient, flexible alignments with support for indels, local alignments, and streaming-based workflows
+
```
Comment (jfy133) [inline docs/bs-seq-primer.md:74]:
```suggestion
```

File: docs/bs-seq-primer.md [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+- By converting reads _in-silico_ on the fly, BWA-Meth eliminates the need for intermediate files, reducing temporary storage requirements and simplifying the overall process
```
Comment (jfy133) [inline docs/bs-seq-primer.md:75]:
```suggestion
- By converting reads _in silico_ on the fly, BWA-Meth eliminates the need for intermediate files, reducing temporary storage requirements and simplifying the overall process
```

File: docs/bs-seq-primer.md [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+
+- Each read set is then aligned using Bowtie2 (alternatively HISAT2 or minimap2) against equally converted references; this enables support for indels, local alignments, and bisulfite converted RNA-seq-type or long reads (e.g. EM-seq using Nanopore or Pac Bio reads)
+
+- By comparing these (up-to) four alignments, Bismark identifies each read’s correct strand origin
+
+This approach allows Bismark to handle directional, PBAT, amplicon, and non-directional libraries robustly and to accurately align reads that represent partially methylated cytosines.
+
+### BWA-Meth ([docs](https://github.com/brentp/bwa-meth); [publication](https://arxiv.org/abs/1401.1129))
+
+- BWA-Meth adapts the BWA-MEM algorithm for bisulfite data, providing efficient, flexible alignments with support for indels, local alignments, and streaming-based workflows
```
Comment (jfy133) [inline docs/bs-seq-primer.md:73]: In what way exactly, what's the technical diff with bismark, given you gave the very nice description above?

File: docs/bs-seq-primer.md [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+
+- By converting reads _in-silico_ on the fly, BWA-Meth eliminates the need for intermediate files, reducing temporary storage requirements and simplifying the overall process
+
+The result is a fast, resource-efficient aligner that integrates smoothly with downstream analysis tools.
+
+## At a glance
```
Comment (jfy133) [inline docs/bs-seq-primer.md:79]: At a glance implies a summary to read before reading the dtails

File: docs/bs-seq-primer.md [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+| **Downstream analysis**               | Built-in methylation calling, filtering and (context-) reporting options                                                                                 | Requires external tools for methylation calling, offering more customization but added complexity                                      |
+| **Recommended when**                  | Accuracy and detailed methylation context are top priority, and ample computational resources are available                                              | Speed (including GPU support), scalability, and modular workflows are essential, or when integrating with existing BWA-based pipelines |
+
+**References**:
+
+- **Bismark [paper](https://pmc.ncbi.nlm.nih.gov/articles/PMC3102221/) and [official docs](https://felixkrueger.github.io/Bismark/)**
+- **BWA-Meth [paper](https://arxiv.org/abs/1401.1129) and [github](https://github.com/brentp/bwa-meth)**
```
Comment (jfy133) [inline docs/bs-seq-primer.md:92]:
```suggestion
- **BWA-Meth [paper](https://arxiv.org/abs/1401.1129) and [GitHub](https://github.com/brentp/bwa-meth)**
```

File: docs/bs-seq-primer.md [hunk 14]
```diff
@@ -0,0 +1,127 @@
...
+
+#### Incomplete conversion and controls:
+
+- **Incomplete bisulfite conversion**: Not all cytosines may be converted to uracil (and eventually to thymine) during the bisulfite treatment step. This partial conversion can lead to overestimation of methylation levels
+- **Spike-In controls**: Often, fully unmethylated lambda DNA or other known-conversion controls are included to estimate and correct for the efficiency of the bisulfite reaction
+
+#### Quality Control (QC) Metrics:
+
+- **M-bias Plots**: Analysis tools can generate M-bias plots to visualize methylation bias across read positions. This helps identify technical artefacts, or non-uniform conversion at biased positiones at the ends of reads
+- **Adapter- and quality trimming**: Even more than standard sequencing data, bisulfite reads benefit from adapter removal and quality filtering. Tools like Trim Galore (often bundled with Bismark) ensure higher-quality alignments and more accurate methylation calls
```
Comment (jfy133) [inline docs/bs-seq-primer.md:105]:
```suggestion
- **Adapter- and quality-trimming**: Even more than standard sequencing data, bisulfite reads benefit from adapter removal and quality filtering. Tools like Trim Galore (often bundled with Bismark) ensure higher-quality alignments and more accurate methylation calls
```

File: docs/bs-seq-primer.md [hunk 15]
```diff
@@ -0,0 +1,127 @@
...
+
+#### Post-alignment deduplication and bias correction:
+
+- **PCR duplication removal**: Bisulfite sequencing libraries often contain PCR duplicates. Removing these duplicates is crucial to avoid artificially inflated methylation signals (relevant for statistics in downstream analysis)
+
+#### Context-specific methylation:
+
+- While the focus often lies on CpG methylation (the most studied context in mammals), accurate mapping and downstream tools can also call methylation in CHG and CHH contexts. This is particularly relevant for plant genomes where non-CpG methylation is biologically significant
```
Comment (jfy133) [inline docs/bs-seq-primer.md:113]: What are CHG/CHH?I don't think you intorudced those acronyms (if you did, ignore this comment)

File: docs/bs-seq-primer.md [hunk 16]
```diff
@@ -0,0 +1,127 @@
...
+| **Downstream analysis**               | Built-in methylation calling, filtering and (context-) reporting options                                                                                 | Requires external tools for methylation calling, offering more customization but added complexity                                      |
+| **Recommended when**                  | Accuracy and detailed methylation context are top priority, and ample computational resources are available                                              | Speed (including GPU support), scalability, and modular workflows are essential, or when integrating with existing BWA-based pipelines |
+
+**References**:
+
+- **Bismark [paper](https://pmc.ncbi.nlm.nih.gov/articles/PMC3102221/) and [official docs](https://felixkrueger.github.io/Bismark/)**
+- **BWA-Meth [paper](https://arxiv.org/abs/1401.1129) and [github](https://github.com/brentp/bwa-meth)**
+- **[NVIDIA Parabricks fq2bam_meth](https://docs.nvidia.com/clara/parabricks/latest/documentation/tooldocs/man_fq2bam_meth.html)**
```
Comment (jfy133) [inline docs/bs-seq-primer.md:93]:
```suggestion
- **[NVIDIA Parabricks `fq2bam_meth`](https://docs.nvidia.com/clara/parabricks/latest/documentation/tooldocs/man_fq2bam_meth.html)**
```

File: docs/bs-seq-primer.md [hunk 17]
```diff
@@ -0,0 +1,127 @@
...
+- While the focus often lies on CpG methylation (the most studied context in mammals), accurate mapping and downstream tools can also call methylation in CHG and CHH contexts. This is particularly relevant for plant genomes where non-CpG methylation is biologically significant
+
+#### Large-scale analysis and cloud computing:
+
+- As whole-genome bisulfite sequencing (WGBS) datasets grow larger, computational efficiency, scalability, and memory usage become critical factors
+- **Cloud-based computing**: Tools that integrate into cloud-based workflows (e.g., AWS, GCP) and employ parallelization or GPU-acceleration (e.g., NVIDIA Parabricks for `fq2bam_meth`) can significantly reduce runtime for large projects
+
+#### Long-read bisulfite sequencing:
+
+- Although short-read Illumina sequencing predominates, emerging methods such as enzymaticallly converted methylation sequencing (EM-seq) are possible using long-read platforms (e.g., PacBio or Oxford Nanopore)
```
Comment (jfy133) [inline docs/bs-seq-primer.md:122]:
```suggestion
- Although short-read Illumina sequencing predominates, emerging methods such as enzymatically converted methylation sequencing (EM-seq) are possible using long-read platforms (e.g., PacBio or Oxford Nanopore)
```

**Review by jfy133 — APPROVED** [review]
Mostly formatting.

Can't speak for accuracy of content, but I learnt something new! This is very nice :D so no blockers

File: docs/bs-seq-primer.md (hunk 3 above)
Comment (maxulysse) [reply docs/bs-seq-primer.md]: Same question

File: docs/bs-seq-primer.md [hunk 18]
```diff
@@ -0,0 +1,132 @@
...
+
+#### Post-alignment deduplication and bias correction:
+
+- **PCR duplication removal**: Bisulfite sequencing libraries often contain PCR duplicates. Removing these duplicates is crucial to avoid artificially inflated methylation signals (relevant for statistics in downstream analysis)
+
+#### Context-specific methylation:
+
+- While the focus often lies on CpG methylation (the most studied context in mammals), accurate mapping and downstream tools can also call methylation in CHG and CHH contexts, which are different sequence environments in which cytosine nucleotides can be found (e.g., “CpG” means a cytosine next to a guanine, while “CHG” and “CHH” indicate cytosines followed by different non-guanine nucleotides), and methylation patterns can vary depending on these contexts. This is particularly relevant for plant genomes where non-CpG methylation is biologically significant.
+
+Note: CpG, CHG, and CHH contexts .
```
Comment (maxulysse) [inline docs/bs-seq-primer.md:118]:
```suggestion
Note: CpG, CHG, and CHH contexts.
```

**Pull request merged**